src/main/java/crawlercommons/sitemaps/SiteMapIndex.java

```diff
@@ -80,8 +80,10 @@ public void addSitemap(AbstractSiteMap sitemap) {
      * @return SiteMap corresponding to the URL or null
      */
     public AbstractSiteMap getSitemap(URL url) {
+        String u = url.toString();
```

Comment:
Can `url` be null here? If so, we should add a check to avoid a NPE

Reply:
... also added a unit test to prove this.